src/components/Preferences/LocalIdentities.js

```diff
@@ -19,132 +20,244 @@ import {
 } from '../IdentityManager/IdentityManager'
 import EmptyLocalIdentities from './EmptyLocalIdentities'
 import Import from './Import'
+import { GU } from '../../utils'
 
 const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream
 
-const LocalIdentities = ({
-  dao,
-  localIdentities,
-  onClearAll,
-  onImport,
-  onModify,
-  onModifyEvent,
-}) => {
-  // transform localIdentities from object into array and attach address to each entry
-  const identities = Object.entries(localIdentities).map(
-    ([address, identity]) => ({
-      ...identity,
-      address,
-    })
-  )
+const SelectableLocalIdentities = React.memo(
+  ({ localIdentities, ...props }) => {
+    const identities = React.useMemo(
+      () =>
+        Object.entries(localIdentities).map(([address, identity]) => ({
+          ...identity,
+          address,
+        })),
+      [localIdentities]
+    )
+    const initialAddressesSelected = React.useMemo(
+      () => new Map(identities.map(({ address }) => [address, true])),
+      [identities]
+    )
+    const [addressesSelected, setAddressesSelected] = React.useState(
+      initialAddressesSelected
+    )
+
+    const handleToggleAll = React.useCallback(
+      () =>
+        setAddressesSelected(
+          new Map(
+            identities.map(({ address }) => [
+              address,
+              !(
+                Array.from(addressesSelected.values()).every(v => v) ||
+                Array.from(addressesSelected.values()).some(v => v)
+              ),
+            ])
+          )
+        ),
+      [addressesSelected, identities]
+    )
+    const handleToggleAddress = React.useCallback(
+      address => () =>
+        setAddressesSelected(
+          new Map([
+            ...addressesSelected,
+            [address, !addressesSelected.get(address)],
+          ])
+        ),
+      [addressesSelected]
+    )
 
-  if (!identities.length) {
-    return <EmptyLocalIdentities onImport={onImport} />
+    React.useEffect(() => {
+      setAddressesSelected(initialAddressesSelected)
+    }, [initialAddressesSelected])
```

Comment:
Do we need this `useEffect()` if the `useState()` starts with `initialAddressesSelected`?

Reply:
Tried without it and it does not work on first render - which actually surprised me.